src/xref2/lookup_failures.ml

```diff
@@ -75,7 +75,7 @@ let report_internal fmt = report ~non_fatal:true fmt
 
 let report_root ~name = add (`Root name)
 
-let report_warning fmt = report ~non_fatal:false fmt
+let report_warning fmt = report ~non_fatal:true fmt
```

Comment:
huh? so warn-error doesn't make _all_ warnings errors? That doesn't make sense, does it?

Reply:
Some warning are never turned into errors indeed.

Reply:
Why?

Reply:
"internal" warnings (most errors from linking) are never turned into fatal errors, as well as failures to lookup root modules.

Reply:
Surely in that case it's an 'info' message rather than a warning, at the very least so people know the meaning of `--warn-error`

Reply:
I think it's because these errors are caused by complicated things and the user can't really fix (bugs in odoc, things expected in the environment that Dune never give).
These things can be fixed but not for 2.0.1.

Reply:
I've made #786 to sort this out, and added it to the 2.1 release project.